packages/pay/src/controllers/rate-probe.ts

```diff
@@ -4,27 +4,29 @@ import {
   StreamRequestBuilder,
   ControllerMap,
   SendState,
-  StreamReject,
   StreamRequest,
   StreamReply,
 } from '.'
-import { Integer } from '../utils'
-import BigNumber from 'bignumber.js'
+import { Int } from '../utils'
 import { MaxPacketAmountController } from './max-packet'
-import { Errors } from 'ilp-packet'
+import { PromiseResolver } from '../utils'
+import { ExchangeRateController, ExchangeRateCalculator } from './exchange-rate'
 
-// TODO What to do otherwise? Use max packet amount?
-// TODO Need algo to discover the **best** exchange rate
-// TODO Add backoff on temporary errors, e.g. T04s?
+// TODO Improve discovery of accurate exchange rates
 
-// TODO Is there a way for this to run passively in the background,
-//      not just during a send loop?
+export interface RateProbeOutcome {
+  maxPacketAmount: Int
+  rateCalculator: ExchangeRateCalculator
+}
 
 export class RateProbe implements StreamController {
-  private disabled = false
+  private static TIMEOUT_MS = 10000
```

Comment:
I think the expiry has to be set differently from the expected duration of the request. Maybe I expect most requests to complete in, e.g., no more than 2x the average latency, but that may not be practical depending upon the min message window subtracted by each connector/how many middleboxes it's passing through. Even 10 seconds for the expiry seems a little short.

Note that when a controller signals to end the payment, the send loop still waits for all pending requests to complete (it just doesn't send any additional packets), so it'd still wait the full 30 seconds to handle any reply packet before exiting, if they end up getting returned. But I still want to fail the probe in that case, since everything took too long.